source/Locales/LawsonLocale/CompactElementsOfPoint.lagda

```diff
@@ -0,0 +1,425 @@
+open import Locales.ScottLocale.Properties pt fe 𝓤
+open import Locales.ScottLocale.ScottLocalesOfAlgebraicDcpos pt fe 𝓤
+open import Locales.ScottLocale.ScottLocalesOfScottDomains pt fe sr 𝓤
+open import Locales.SmallBasis pt fe sr
+open import Locales.Spectrality.SpectralMap pt fe
+open import Locales.TerminalLocale.Properties pt fe sr
+open import Slice.Family
+open import UF.Logic
+open import UF.Powerset
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier renaming (⊥ to ⊥ₚ)
+open import UF.Univalence
+
+open AllCombinators pt fe renaming (_∧_ to _∧ₚ_; _∨_ to _∨ₚ_)
+open FrameHomomorphismProperties
+open FrameHomomorphisms
+open Locale
+open PropositionalTruncation pt hiding (_∨_)
+
+\end{code}
+
+We work in a module parameterized by a large and locally small locale `X`.
+
+\begin{code}
+
+module Preliminaries (X : Locale (𝓤 ⁺) 𝓤 𝓤) where
+
+ open ContinuousMaps
+
+\end{code}
+
+We use the abbreviation `𝟏L` for the terminal locale of the category of
+`𝓤`-locales (i.e. large and locally small locales over universe `𝓤`).
+
+\begin{code}
+
+ 𝟏L : Locale (𝓤 ⁺) 𝓤 𝓤
+ 𝟏L = 𝟏Loc pe
+
+\end{code}
+
+For the reader who might not be familiar, this is the locale defined by the
+frame of opens `Ω`.
+
+\begin{code}
+
+ _ : ⟨ 𝒪 𝟏L ⟩ ＝ Ω 𝓤
+ _ = refl
+
+\end{code}
+
+By a point of locale, we mean a continuous map from `𝟏L` into `X`.
+
+\begin{code}
+
+ Point : 𝓤 ⁺  ̇
+ Point = 𝟏L ─c→  X
+
+\end{code}
+
+We now proceed to the definition of the family mentioned in the preamble. We
+work with a dcpo `𝓓` that is assumed to
+
+  - have a bottom element,
+  - be a Scott domain, and
+  - satisfy the aforementioned decidability condition for upper boundedness.
+
+\begin{code}
+
+open DefinitionOfScottDomain
+
+module Construction
+        (𝓓    : DCPO {𝓤 ⁺} {𝓤})
+        (ua   : Univalence)
+        (hl   : has-least (underlying-order 𝓓))
+        (sd   : is-scott-domain 𝓓 holds)
+        (dc   : decidability-condition 𝓓) where
+
+ open SpectralScottLocaleConstruction₂ 𝓓 ua hl sd dc pe
+ open SpectralScottLocaleConstruction 𝓓 hl hscb dc bc pe hiding (scb; β; ϟ)
+ open DefnOfScottTopology 𝓓 𝓤
+
+ open Preliminaries σ⦅𝓓⦆
+ open Properties 𝓓
+
+\end{code}
+
+We denote by `B𝓓` the basis of `𝓓`.
+
+\begin{code}
+
+ B𝓓 : Fam 𝓤 ⟨ 𝓓 ⟩∙
+ B𝓓 = index-of-compact-basis 𝓓 hscb , family-of-compact-elements 𝓓 hscb
+
+\end{code}
+
+We use the abbreviation `scb` for the proof that `B𝓓` is a small basis
+consisting of compact opens.
+
+\begin{code}
+
+ scb : is-small-compact-basis 𝓓 (family-of-compact-elements 𝓓 hscb)
+ scb = small-compact-basis 𝓓 hscb
+
+\end{code}
+
+By `βₖ i`, we denote the element denoted by an index `i`, packaged up with the
+proof that it is compact.
+
+\begin{code}
+
+ open is-small-compact-basis scb
+ open BottomLemma 𝓓 𝕒 hl
+
+ βₖ : (i : index B𝓓) → Σ c ꞉ ⟨ 𝓓 ⟩∙ , is-compact 𝓓 c
+ βₖ i = B𝓓 [ i ] , basis-is-compact i
+
+\end{code}
+
+We now write down the family of compact opens of a point which we denote
```

Comment:
So first, I have a terrible habit of saying compact _open_ instead of compact _element_ 😆 This is one thing that is confusing here. I’ll be more careful about this in the future.

The thing I am trying say in this comment is that this is the family of compact elements of the domain whose principal filters fall in the given point. So it makes sense to think of this as the family of compact approximants of _the point_ in consideration.

Honestly, the geometric intuition here is not entirely clear to me and I don’t know if there’s a standard name for this notion in the literature. I don’t like the name `𝒦-in-point`. Please let me know if you have any suggestions. I want to improve this name and this comment.

When I wrote this proof on paper, I intuitively thought of this as the family of compact neighbourhoods converging around the point. But I’m not entirely sure if this geometric view is accurate.

Reply:
I would just (mostly) repeat your second paragraph here as a comment in the file.

Reply:
I have now added the following explanation:

> Ideally, the name here would be `𝒦-with-principal-ideals-in-point` but this is too long, which is why we use the name `𝒦-in-point`.

> It makes sense to me to think of this as the family of compact approximants to the given point, but I'm not sure this geometric view is accurate at the time of writing. I will improve this name in the future as my understanding of the underlying geometric intuition increases.